xbmc/filesystem/NFSFile.cpp

```diff
@@ -351,11 +351,11 @@ void CNfsConnection::Deinit()
 /* This is called from CApplication::ProcessSlow() and is used to tell if nfs have been idle for too long */
 void CNfsConnection::CheckIfIdle()
 {
+  CSingleLock lock(*this);
   /* We check if there are open connections. This is done without a lock to not halt the mainthread. It should be thread safe as
    worst case scenario is that m_OpenConnections could read 0 and then changed to 1 if this happens it will enter the if wich will lead to another check, wich is locked.  */
   if (m_OpenConnections == 0 && m_pNfsContext != NULL)
   { /* I've set the the maxiumum IDLE time to be 1 min and 30 sec. */
-    CSingleLock lock(*this);
```

Comment:
See the comment above that change. It was ugly and not thread safe before, but the author knew what he did.